### DISCUSSION TITLE
Anyone successfully used Unistyles in a monorepo?

### DISCUSSION THREAD

**Discussion by atanaskanchev** [jpudysz/react-native-unistyles#133] [Q&A]
So far my turborepo setup looks like

-- apps/native
-- packages
- unistyles
- theme
- components

In order to have the theme registered in the app I've added `import '@theapp/packages-unistyles'` in the `apps/native/app/_layout.tsx` and that works ok.

In order to use the type for the theme in the components package I need to add `import '@theapp/packages-unistyles'` to every file.

Did someone manage to implement a better monorepo structure?

**Comment by jpudysz** [comment]
Please check this repo with turborepo and solito:
https://github.com/jpudysz/react-native-unistyles-ssr-exmaple

**Reply by atanaskanchev** [reply]
Thanks for the pointer, however in the solito example the components and the initialisation of unistyles happens in the same package. In the case where we want to spilt the components out in another package, the theme registers as type `never`. Importing the unitsyles init file solves the types issue, but it has to be added to every file. I suppose there isn't a performance overhead doing this?

**Reply by jpudysz** [reply]
That doesn't seems to be right. Can you give ma a small repo with your example so I will be able to play with it and find solution?

**Reply by jpudysz** [reply]
Any update? I'm willing to help, but I need your input :)

**Comment by lobor** [comment]
Here a monorepo with unistyles
https://github.com/Paymium/crossed

the typescript init in this file https://github.com/Paymium/crossed/blob/main/packages/ui/src/theme/index.ts
a new PR will create tomorrow with better integration in apps/documentation for next app router

**Comment by erickreutz** [comment]
@lobor why did you rip out unistyles?

**Reply by lobor** [reply]
I don't know why we deleted it, we used unistyles 2
